src/include/gcc.h

```diff
@@ -282,8 +259,6 @@ WT_ATOMIC_FUNC(size, size_t, size_t *vp, size_t v)
         __asm__ volatile("bcr 15,0\n" ::: "memory"); \
     } while (0)
 #define WT_READ_BARRIER() WT_FULL_BARRIER()
-/* We only need a compiler barrier for s390x as its memory ordering is strong enough. */
```

Comment:
This is weird. The `WEAK_MEMORDER` comment indicates that `WT_READ_BARRIER` should be a compiler barrier, but we actually use a full barrier? It's the same for i386.
Not a blocker for this PR but it warrants a ticket outside the scope of PM-3221

Reply:
I can create a follow up ticket for this. I don't think its a big deal as its not one of our main architectures.

Reply:
WT-12201